test/utils/rollout_utils_test.ts

```diff
@@ -0,0 +1,46 @@
+// tslint:disable:custom-no-magic-numbers
+// tslint:disable: boolean-naming
+
+import { expect } from '@0x/contracts-test-utils';
+
+import { isRolledOut } from '../../src/utils/rollout_utils';
+
+describe('isRolledOut', () => {
+    it('should give a consistent result for the same message', () => {
+        const message = 'hello world';
+        const threshold = 0.5;
+
+        const first = isRolledOut({ message, threshold });
+        const second = isRolledOut({ message, threshold });
+        const third = isRolledOut({ message, threshold });
+
+        expect(first).to.eq(second);
+        expect(second).to.eq(third);
+    });
+
+    it('should partition the rollout group to the approximate threshold target', () => {
+        const population = 1000;
+        const threshold = 0.1;
+        let rolloutCount = 0;
+
+        for (let i = 0; i < population; i++) {
+            if (isRolledOut({ message: i.toString(), threshold })) {
+                rolloutCount++;
+            }
+        }
+        expect(rolloutCount).to.eq(98); // approximately 100
```

Comment:
can use this for approximate things https://jestjs.io/docs/expect#expectclosetonumber-numdigits

Reply:
oh nm we're in non jest land